### PR TITLE
subsys: Bluetooth: df: Add reporting of insufficient resources for IQ samples

### PR DESCRIPTION
**Pull request by ppryga-nordic**
According to Bluetooth 5.3 Vol 4, Part E section 7.7.65.21 the
Controller should send an IQ report with Packet_Status set to 0xFF
and Sample_Count to 0x0, to inform Host that it is not able to
store IQ samples for all received Constant Tone Extensions and has
failed to sample at least once.

This commit adds handling of such situation for direction finding
in connectionless mode in lower link layer. It includes:
- lack of node_rx_iq_report to store IQ samples,
- lack of free node_rx to report received advertising data (generates
  an advertising data report with indication "incomplete, no more to
  come" and cancels reception of chained PDUs,
- cancellation of chained PDUs reception due to other scheduling
  reasons.
In all of these situations CTE reception must be enabled and number of
received CTEs is smaller than requested by Host.

If there is not sufficient number of node_rx_iq_report nodes and
report was generated, the CTE will not be samples in a given
periodic advertising synchronization event. The CTE will be
sampled again in next event if there are node_rx_iq_report available.

End user applications may check periodic advertising sync event
counter when use of the reports to be sure about sync event
the received report is connected with.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

### CODE REVIEW

**Comment by ppryga-nordic** [comment]
Fixed commit message to pass compliance. 
Rebased onto latest main.

**Comment by ppryga-nordic** [comment]
Added more information about implemented enhancement to commit message.

**Comment by ppryga-nordic** [comment]
@cvinayak can I close this PR? Or are you going to do something with it?

**Comment by cvinayak** [comment]
@ppryga-nordic I will do the changes, I need reviewers. This PR needs to be merged upstream.

File: subsys/bluetooth/controller/ll_sw/ull.c [hunk 1]
```diff
@@ -448,12 +448,10 @@ static struct {
...
-#define LINK_RX_POOL_SIZE                                                      \
-	(sizeof(memq_link_t) *                                                 \
-	 (RX_CNT + 2 + BT_CTLR_MAX_CONN + BT_CTLR_ADV_SET +                    \
-	  (BT_CTLR_ADV_ISO_SET * 2) + (BT_CTLR_SCAN_SYNC_SET * 2) +            \
-	  (BT_CTLR_SCAN_SYNC_ISO_SET * 2) +                                    \
-	  (IQ_REPORT_CNT)))
+#define LINK_RX_POOL_SIZE                                                                          \
+	(sizeof(memq_link_t) *                                                                     \
+	 (RX_CNT + 2 + BT_CTLR_MAX_CONN + BT_CTLR_ADV_SET + (BT_CTLR_ADV_ISO_SET * 2) +            \
+	  (BT_CTLR_SCAN_SYNC_SET * 2) + (BT_CTLR_SCAN_SYNC_ISO_SET * 2) + (IQ_REPORT_CNT)))
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull.c]: Remove the change

File: subsys/bluetooth/controller/ll_sw/ull_df.c [hunk 2]
```diff
@@ -66,21 +66,33 @@
 #define IQ_REPORT_STRUCT_OVERHEAD  (IQ_REPORT_HEADER_SIZE)
 #define IQ_SAMPLE_SIZE (sizeof(struct iq_sample))
 
-#define IQ_REPORT_RX_NODE_POOL_ELEMENT_SIZE              \
+#define IQ_REPORT_RX_NODE_POOL_ELEMENT_SIZE                                                        \
 	MROUND(IQ_REPORT_STRUCT_OVERHEAD + (IQ_SAMPLE_TOTAL_CNT * IQ_SAMPLE_SIZE))
-#define IQ_REPORT_POOL_SIZE (IQ_REPORT_RX_NODE_POOL_ELEMENT_SIZE * IQ_REPORT_CNT)
+#define IQ_REPORT_POOL_SIZE                                                                        \
+	(IQ_REPORT_RX_NODE_POOL_ELEMENT_SIZE * IQ_REPORT_CNT)
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_df.c]: Removed the change

**Review by cvinayak — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
@carlescufi can you we have a second approval?

**Review by carlescufi — APPROVED** [review]

**Pull request merged**